etna/pipeline/autoregressive_pipeline.py

```diff
@@ -130,8 +136,23 @@ def _forecast(self) -> TSDataset:
                     message="You probably set wrong freq.",
                     action="ignore",
                 )
-                current_ts_forecast = current_ts.make_future(current_step)
-            current_ts_future = self.model.forecast(current_ts_forecast)
+
+                if isinstance(self.model, get_args(ContextRequiredModelType)):
+                    self.model = cast(ContextRequiredModelType, self.model)
+                    if isinstance(self.model, DeepBaseModel):
+                        current_ts_forecast = current_ts.make_future(
+                            future_steps=self.model.decoder_length, tail_steps=self.model.encoder_length
```

Comment:
self.model.context_size $\neq$ self.model.encoder_length ?

Reply:
For `DeepBaseModel` they are the same.

Reply:
Why do we use extra branch in this case?

Reply:
Because of `self.model.decoder_length`.

Reply:
Ok, we should fix it in the future